diff --git a/Marlin/Configuration.h b/Marlin/Configuration.h
index 6b0d8acaee..66abf6b25e 100644
--- a/Marlin/Configuration.h
+++ b/Marlin/Configuration.h
@@ -126,11 +126,11 @@
 // Enable the Bluetooth serial interface on AT90USB devices
 //#define BLUETOOTH
 
 // Choose the name from boards.h that matches your setup
 #ifndef MOTHERBOARD
-  #define MOTHERBOARD BOARD_CREALITY_V452 // BOARD_CREALITY_V453
+  #define MOTHERBOARD BOARD_CREALITY_V453 // BOARD_CREALITY_V452
 #endif
 
 // Name displayed in the LCD "Ready" message and Info menu
 #define CUSTOM_MACHINE_NAME "CR-6 Community"
 
@@ -1159,20 +1159,20 @@
 #define Z_HOME_DIR -1
 
 // @section machine
 
 // The size of the print bed
-#define X_BED_SIZE 235
-#define Y_BED_SIZE 235
+#define X_BED_SIZE 400
+#define Y_BED_SIZE 400
 
 // Travel limits (mm) after homing, corresponding to endstop positions.
-#define X_MIN_POS -5
-#define Y_MIN_POS -2
+#define X_MIN_POS -10
+#define Y_MIN_POS -3
 #define Z_MIN_POS 0
-#define X_MAX_POS X_BED_SIZE
-#define Y_MAX_POS Y_BED_SIZE
-#define Z_MAX_POS 250
+#define X_MAX_POS 410
+#define Y_MAX_POS 404
+#define Z_MAX_POS 405
 
 /**
  * Software Endstops
  *
  * - Prevent moves outside the set machine bounds.
@@ -1372,11 +1372,11 @@
 #endif
 
 #if EITHER(AUTO_BED_LEVELING_LINEAR, AUTO_BED_LEVELING_BILINEAR)
 
   // Set the number of grid points per dimension.
-  #define GRID_MAX_POINTS_X 4
+  #define GRID_MAX_POINTS_X 7
   #define GRID_MAX_POINTS_Y GRID_MAX_POINTS_X
 
   // Probe along the Y axis, advancing X after each column
   #define PROBE_Y_FIRST
 
